docs/adr/0014-on-chain-state-management.md

```diff
@@ -0,0 +1,162 @@
+
+> Note that the `Init` transaction is the most complicated transition to observe for there's no previous state. For other transitions such as `CollectCom` or `Abort`, one can get away with solely observing that the head script `νHead` is executing the expected transition; the script validator is already checking any other details. 
+
+From observing an `Init` transaction, observers may infer the following pieces of information:
+
+- The head parameters, such as the contestation period;
+- Ephemeral verification keys of other parties;
+- A UTXO entry locked by the `νHead` script;
+- Some UTXO entries locked by `νInitial` scripts.
+
+#### Observing a 'Commit' transaction
+
+A transaction qualifies as a `Commit` if and only if
+
+- There's exactly one input spending funds locked by a script `νInitial` with a `Commit` redeemer.
+
+From observing a `Commit` transaction, observers may infer the following pieces of information:
+
+- A UTXO entry locked by a `νCommit` script;
+- A now-spent-UTXO entries which was locked by a `νInitial` script.
+
+#### Observing an 'Abort' transaction
+
+A transaction qualifies as an `Abort` if and only if:
+
+- There's exactly one input spending funds locked by a script `νHead` in an `Initial` state, using an `Abort` redeemer. 
+
+From observing an `Abort` transaction, observers may infer the following pieces of information:
+
+- A now-spent-UTXO entry which were locked by `νHead` script;
+- Some now-spent-UTXO entries which were locked by `νInitial` scripts;
+- Some now-spent-UTXO entries which were locked by `νCommit` scripts.
+
+#### Observing a 'CollectCom' transaction
+
+A transaction qualifies as a `CollectCom` if and only if:
+
+- There's exactly one input spending funds locked by a script `νHead` in an `Initial` state, using a `CollectCom` redeemer. 
+
+From observing a `CollectCom` transaction, observers may infer the following pieces of information:
+
+- A new UTXO entry locked by a `νHead` script;
+- Some now-spent-UTXO entries which were locked by `νCommit` scripts;
+- An initial UTXO set (and a corresponding root hash for that UTXO set).
+
+#### Observing a 'Close' transaction
+
+A transaction qualifies as a `Close` if and only if:
+
+- There's exactly one input spending funds locked by a script `νHead` in an `Open` state, using a `Close` redeemer. 
+
+From observing a `Commit` transaction, observers may infer the following pieces of information:
+
+- A new UTXO entry locked by a `νHead` script;
+- A candidate final UTXO set
+- A candidate snapshot number 
+
+#### Observing a 'Fanout' transaction
+
+A transaction qualifies as a `Fanout` if and only if:
+
+- There's exactly one input spending funds locked by a script `νHead` in a `Closed` state, using a `Fanout` redeemer. 
+
+From observing a `Commit` transaction, observers may infer the following pieces of information:
+
+- A candidate final UTXO set
+- A candidate snapshot number 
+
```

Comment:
Not reviewed all details of above statements as they feel a bit out of place here or at least I don't see why that's relevant for the decision proposed. Maybe move these into the code base as actual documentation/haddocks?